surge_test.go

```diff
@@ -2,425 +2,289 @@ package surge_test
 
```

Comment:
Generic comment on testing: I don't think I saw any tests that marshalled and then unmarshalled some random types/structs and checked that the initial and unmarshaled types/structs were the same. I think these kind of tests are important.

Reply:
These tests do exist for the more complex types, but that is a good point. I will add them for the simpler types too.